endpoint/endpoint.go

```diff
@@ -191,10 +192,52 @@ func (e *Endpoint) GetProviderSpecificProperty(key string) (ProviderSpecificProp
 	return ProviderSpecificProperty{}, false
 }
 
+type NormalizeProviderSpecificConfig map[string]func(string) string
+
+// Selects and normalizes fields specified by config
+func (e *Endpoint) NormalizeProviderSpecific(config NormalizeProviderSpecificConfig) {
+	normalized := ProviderSpecific{}
+
+	for name, normalize := range config {
+		if property, ok := e.GetProviderSpecificProperty(name); ok {
+			normalized = append(normalized, ProviderSpecificProperty{
+				Name:  name,
+				Value: normalize(property.Value),
+			})
+		} else {
+			normalized = append(normalized, ProviderSpecificProperty{
+				Name:  name,
+				Value: normalize(""),
+			})
+		}
+	}
+
+	e.ProviderSpecific = normalized
+}
+
 func (e *Endpoint) String() string {
 	return fmt.Sprintf("%s %d IN %s %s %s %s", e.DNSName, e.RecordTTL, e.RecordType, e.SetIdentifier, e.Targets, e.ProviderSpecific)
 }
 
+// Normalizes boolean value of custom property
+func NormalizeBoolean(value string, defaultValue bool) string {
+	val, err := strconv.ParseBool(value)
+
+	if err != nil {
+		if defaultValue {
+			return "true"
+		} else {
+			return "false"
+		}
+	}
+
+	if val {
+		return "true"
+	}
+
+	return "false"
+}
+
```

Comment:
Why do we need this to be part of `endpoint.go`? Also, using strings with `"true"` is always a hard sell. I understand you tried to stick to a `func(string) string` as function, but this is hard do grasp IMO.

Reply:
where would you put it? `"true"` is what is used currently. using anything else would result in unnecessary update of DNS

Reply:
Custom annotations have format map[string]string. Result type must be string, it's the point of this function :)